andela/haven-ah-backend
# #161342835 Create functionality to send JWT on signup

#### What does this PR do?
- Create signup route.

#### Description of Task to be completed?
- Implement JWT for user authentication on successful signup
- Implement password hashing.
- Update user model to include `firstName` and `lastName`

#### How should this be manually tested?
- git clone `https://github.com/andela/haven-ah-backend.git`.
- Install `node` if not installed.
- ensure that PostgreSQL is running on your local machine.
- run `npm install` to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman
- run `npm run start-dev`
- on postman navigate to `localhost:5000/api/users/signup`.

**Sample Request Body** 
```gherkhin
{
  "username": "string",
  "firstName": "string",
  "lastName": "string",
  "email": "string",
  "password": "string",
}
```

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#161342835

#### Screenshots (if appropriate)
<img width="1091" alt="screen shot 2018-10-26 at 10 54 52 am" src="https://user-images.githubusercontent.com/29674347/47559558-bdfbd400-d90d-11e8-88c6-edaafa27b7f5.png">

<img width="1080" alt="screen shot 2018-10-26 at 10 55 13 am" src="https://user-images.githubusercontent.com/29674347/47559525-aa506d80-d90d-11e8-9d6b-ded2f728d8b2.png">
